prow/config/secret/agent.go

```diff
@@ -128,17 +136,24 @@ func (a *Agent) GetTokenGenerator(secretPath string) func() []byte {
 	}
 }
 
-const censored = "CENSORED"
-
-var censoredBytes = []byte(censored)
-
 // Censor replaces sensitive parts of the content with a placeholder.
 func (a *Agent) Censor(content []byte) []byte {
-	for sKey := range a.secretsMap {
-		secret := a.GetSecret(sKey)
-		content = bytes.ReplaceAll(content, secret, censoredBytes)
+	out := make([]byte, len(content))
+	copy(out, content)
+	return a.CensorInPlace(out)
+}
+
+// CensorInPlace replaces sensitive parts of the content with a placeholder,
+// mutating the input content.
+func (a *Agent) CensorInPlace(content []byte) []byte {
```

Comment:
Is there any usecase where we want this? I would prefer not to have two seperate `Censor` methods in the agents api. Also, lets not return something, that makes misuse too easy as it creates the impression that was a copy.

Reply:
We want this always? I don't see when we *don't* want this.

Reply:
Then lets remove the old Censor?

Reply:
People vendor this code. I want to not break them :)